src/lyra.ts

```diff
@@ -648,15 +649,22 @@ export function search<S extends PropertiesSchema>(
     for (let j = 0; j < tokensLength; j++) {
       const term = tokens[j];
       const documentIDs = getDocumentIDsFromSearch(lyra, { ...params, index, term, exact });
-      const termOccurrencies = lyraOccurrencies[term];
+      
+      // lyraOccurrencies[term] can be undefined, 0, string, or { [k: string]: number }
+      const termOccurrencies = typeof lyraOccurrencies[term] === "number"
+        ? (lyraOccurrencies[term] ?? 0)
+        : 0;
+
       const orderedTFIDFList: TokenScore[] = [];
 
       // Calculate TF-IDF value for each term, in each document, for each index.
       // Then insert sorted results into orderedTFIDFList.
       const documentIDsLength = documentIDs.length;
       for (let k = 0; k < documentIDsLength; k++) {
+        // idf's denominator is shifted by 1 to avoid division by zero
+        const idf = Math.log10(N / (1 + termOccurrencies));
```

Comment:
This is such a common scenario that even wikipedia mentions this "unit shift" trick ([link](https://en.wikipedia.org/wiki/Tf%E2%80%93idf#Inverse_document_frequency_2)):

> If the term is not in the corpus, this will lead to a division-by-zero. It is therefore common to adjust the denominator.

Reply:
This implies that the TF-IDF scores are now slightly smaller in the snapshot tests.

Reply:
That's right. Thanks a million for that, we used to add `1` and eventually removed it for testing purposes, but never re-introduced it again